pkg/osd_cluster_upgrader/upgrader.go

```diff
@@ -641,9 +641,15 @@ func performUpgradeFailure(metricsClient metrics.Metrics, nc eventmanager.EventM
 	// flag window breached metric
 	metricsClient.UpdateMetricUpgradeWindowBreached(upgradeConfig.Name)
 
+	// TearDown the extra machineset
```

Comment:
I think it may be better if this happens at the start of the function. That way, if the scale down were to somehow fail and retry on the next reconcile, it isn't attempting to resend failure notifications (which immediately triggers a policy cancellation/deletion on the OCM end) and overwriting the window breach metric. The successful scale-down should be a prerequisite before the notification of failure to OCM